compute/src/main/java/org/zstack/compute/vm/VmInstanceApiInterceptor.java

```diff
@@ -146,31 +146,28 @@ protected void scripts() {
                 VmInstanceVO vo = Q.New(VmInstanceVO.class).eq(VmInstanceVO_.uuid, msg.getVmInstanceUuid()).find();
                 Integer cpuSum = msg.getCpuNum();
                 Long memorySize = msg.getMemorySize();
-                if (cpuSum == null && memorySize == null) {
+                if ((cpuSum == null && memorySize == null)) {
                     return;
                 }
 
+                VmInstanceState vmState = Q.New(VmInstanceVO.class).select(VmInstanceVO_.state).eq(VmInstanceVO_.uuid, msg.getVmInstanceUuid()).findValue();
+                if (!VmGlobalConfig.NUMA.value(Boolean.class) && !VmInstanceState.Stopped.equals(vmState)) {
+                    throw new ApiMessageInterceptionException(argerr(
+                            "cannot do cpu or memory update because numa not allowed when vm not in the state of %s", vmState
```

Comment:
"cannot do cpu or memory update because numa not allowed when vm not in the state of %s" --> "the VM cannot do online cpu/memory update because it is not of NUMA architecture. Please stop the VM then do the cpu/memory update again"